plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt

```diff
@@ -79,7 +80,42 @@ abstract class LoginBrowser(
 
     protected val selectionSettings = mutableMapOf<String, BearerConnectionSelectionSettings>()
 
-    protected val onPendingToken: (InteractiveBearerTokenProvider) -> Unit = { provider ->
+    private var browserOpenTimer: Timer? = null
+
+    private fun startBrowserOpenTimer(credentialSourceId: CredentialSourceId) {
+        browserOpenTimer = Timer()
+        browserOpenTimer?.schedule(
+            object : TimerTask() {
+                override fun run() {
+                    AwsTelemetry.loginWithBrowser(
+                        project = null,
+                        credentialStartUrl = SONO_URL,
+                        result = Result.Failed,
+                        reason = "Browser authentication idle for more than 15min",
+                        credentialSourceId = credentialSourceId
+                    )
+                    AuthTelemetry.addConnection(
+                        result = Result.Failed,
+                        reason = "Browser authentication idle for more than 15min",
+                        credentialSourceId = credentialSourceId
+                    )
+                    stopBrowserOpenTimer()
```

Comment:
if this task only runs once, why do we need to cancel it?

Reply:
We don't need to cancel it here. The relevant logic from `stopBrowserOpenTimer()` is setting `browserOpenTimer` to null